### DISCUSSION TITLE
Midi In (RX) messages bridging to Midi Out (TX)

### DISCUSSION THREAD

**Discussion by RiNoize** [FortySevenEffects/arduino_midi_library#246] [Q&A] [answered]
Hello franky, I hope you are well.
I want to add a MIDI in to my controller (I understand that it requires a circuit with an optocoupler connected to the RX pin of the arduino) to plug in another Midi controller, and that information pass as through a bridge (like an internal MIDI Thru) to the serial Midi out (TX) without any kind of filtering.

For example:
With my controller I am generating midi messages on channels 1, 10 and 16, and with the external controller plugged into the Midi in I am generating midi messages on channel 2.
So I want to have all those midi messages coming out on TX, channels 1, 2, 10 and 16.
How should I implement that in the sketch?
Thank you in advance.

Richard

**Comment by franky47** [comment, marked as answer]
Hi Richard,

That's a one-line answer: add `MIDI.turnThruOn();` after `MIDI.begin();` in your `setup` function.

Your controller will continue transmitting its messages as usual, and incoming data will be mirrored to the TX pin.

**Reply by RiNoize** [reply]
Gracias Franky
